src/ChunkDataCallback.h

```diff
@@ -101,3 +101,37 @@ class cChunkDataSeparateCollector :
 
 
 
+/** A simple implementation of the cChunkDataCallback interface that just copies the cChunkData */
+class cChunkDataCopyCollector :
+	public cChunkDataCallback
+{
+public:
+	struct MemCallbacks:
+		cAllocationPool<cChunkData::sChunkSection>::cStarvationCallbacks
+	{
+		virtual void OnStartUsingReserve() override {}
+		virtual void OnEndUsingReserve() override {}
+		virtual void OnOutOfReserve() override {}
+	};
+
+	cChunkDataCopyCollector():
+		m_Pool(cpp14::make_unique<MemCallbacks>()),
+		m_Data(m_Pool)
+	{
+	}
+
+
+	cListAllocationPool<cChunkData::sChunkSection, cChunkData::NumSections> m_Pool;  // Keep 1 chunk worth of reserve
```

Comment:
`cListAllocationPool` isn't thread safe so this _has_ to have it's own pool.  I could maybe add a new type of pool that uses a static buffer though.